code/game/machinery/computer/dna_console.dm

```diff
@@ -1357,9 +1383,16 @@
 			if(!can_modify_occupant())
 				return
 
-			// Set the appropriate timer and index to pulse. This is then managed
+			// Set the appropriate timer, string, and index to pulse. This is then managed
 			//  later on in process()
-			var/len = length_char(scanner_occupant.dna.uni_identity)
+			var/type = params["type"]
+			rad_pulse_type = type
+			var/len
+			switch(type)
+				if("ui")
+					len = length_char(scanner_occupant.dna.unique_identity)
```

Comment:
Do these strings use any non-ASCII characters? If they don't, simply use `length()`, as it's cheaper.

Reply:
Only through adminbus, but it was a length_char call before this PR. I can change it, though.